### PR TITLE
Add support for "type" key in JSON text

### PR DESCRIPTION
**Pull request by Nico314159**

### CODE REVIEW

**Review by MulverineX — CHANGES_REQUESTED** [review]
There should be a literal `type` defined for each of the `TextObject` structs, even if we did use a single `type` it would go in `TextBase`, not `TextStyle`. `TextStyle` is used for other things like `/team`

**Comment by Nico314159** [comment]
> There should be a literal `type` defined for each of the `TextObject` structs, even if we did use a single `type` it would go in `TextBase`, not `TextStyle`. `TextStyle` is used for other things like `/team`

`/team` can have clickEvent in it?

File: java/server/util/text.mcdoc [hunk 1]
```diff
@@ -10,45 +10,61 @@ type Text = (
 type TextObject = (
 	struct {
 		text: string,
+		#[since="1.20.3"]
+		type?: "string",
```
Comment (MulverineX) [inline java/server/util/text.mcdoc:14]: this is `text` not `string`

**Review by MulverineX — APPROVED** [review]

**Review by MulverineX — CHANGES_REQUESTED** [review]
whoops

**Review by MulverineX — APPROVED** [review]

**Pull request merged**